qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java

```diff
@@ -0,0 +1,209 @@
+import io.camunda.zeebe.test.util.actuator.ClockActuatorClient;
+import io.camunda.zeebe.test.util.testcontainers.ZeebeTestContainerDefaults;
+import io.zeebe.containers.ZeebeContainer;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.agrona.CloseHelper;
+import org.elasticsearch.client.RestClient;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.Network;
+import org.testcontainers.elasticsearch.ElasticsearchContainer;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.shaded.org.awaitility.Awaitility;
+import org.testcontainers.utility.DockerImageName;
+
+@Testcontainers
+public class ControlledActorClockEndpointIT {
+  private static final String ELASTICSEARCH_HOST = "elasticsearch";
+  private static final String INDEX_PREFIX = "exporter-clock-test";
+  private Network network;
+  private ElasticsearchContainer elasticsearchContainer;
+  private ZeebeContainer zeebeContainer;
+  private ZeebeClient zeebeClient;
+  private final HttpClient httpClient = HttpClient.newHttpClient();
+  private final ObjectMapper mapper = new ObjectMapper();
+
+  private ClockActuatorClient clockActuatorClient;
+
+  @BeforeEach
+  void startContainers() {
+    network = Network.newNetwork();
+    startElasticsearch();
+    startZeebe();
+  }
+
+  @AfterEach
+  void stopContainers() {
+    CloseHelper.closeAll(zeebeClient, zeebeContainer, elasticsearchContainer, network);
+  }
+
+  @Test
+  void testPinningTime() throws IOException, InterruptedException {
+    // given - Zeebe actor clock is pinned
+    final var pinnedAt = clockActuatorClient.pinZeebeTime(Instant.now().minus(Duration.ofDays(3)));
+    final var process = Bpmn.createExecutableProcess().startEvent().endEvent().done();
+
+    // when - producing records
+    final var deployResult =
+        zeebeClient.newDeployCommand().addProcessModel(process, "process.bpmn").send().join();
+    zeebeClient
+        .newCreateInstanceCommand()
+        .processDefinitionKey(
+            deployResult.getProcesses().stream()
+                .findFirst()
+                .orElseThrow()
+                .getProcessDefinitionKey())
+        .send()
+        .join();
+
+    // then - records are exported with a timestamp matching the pinned time
+    Awaitility.await()
+        .untilAsserted(
+            () -> {
+              final var records = searchExportedRecords();
+              assertThat(records)
+                  .hasSize(10) // deploy + instance produces exactly 10 records
```

Comment:
:wrench: This is coupling the test to the output of the engine. A think slightly more stable version would be to just deploy a process, then await until the number of records doesn't change (i.e. the engine is idle), and compare them. Same for the test below.